R/permissions.R

```diff
@@ -0,0 +1,50 @@
+# @file permissions
+#
+# Copyright 2021 Observational Health Data Sciences and Informatics
+#
+# This file is part of ROhdsiWebApi
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#' Assign cohort definition permissions to user.
+#'
+#' @details
+#' The function will call the webservice to assign permissions to a cohort definition for the given userId.
+#'
+#' @param baseUrl     The URL to WebAPI
+#' @param cohortId    The ID of the cohort definition
+#' @param userId      The userId to grant permission
+#' @param permission  The permission type to assign.  Can be READ or WRITE. Defaults to "WRITE".    
+#'
+#' @examples
+#' \dontrun{
+#' getCohortSql(CohortDefinition = (getCohortDefinition(cohortId = 13242, baseUrl = baseUrl)),
+#'              baseUrl = "http://server.org:80/WebAPI")
+#' }
+#' @export
+addCohortPermission <- function(baseUrl, cohortId, userId, permission="WRITE") {
```

Comment:
Does this function work with older versions of atlas (i.e. is there an informative error if the permission endpoints don't exist?)

Reply:
Not exactly: the problem is that there was a API breaking change in atlas 2.10 related to the call to `check`, so killing two birds here:  even with errors, you should be allowed to save cohorts, and b) removing a call to `check` that fails tests.  I can remove this code block if it's distracting, but I figured I'd just call it out.